block-storage/src/main/scala/coop/rchain/blockstorage/InMemBlockStore.scala

```diff
@@ -25,20 +25,22 @@ class InMemBlockStore[F[_]] private ()(
     for {
       _     <- metricsF.incrementCounter("get")
       state <- refF.get
-    } yield state.get(blockHash)
+    } yield state.get(blockHash) >>= (bmp => BlockMessage.from(bmp).toOption)
 
   override def find(p: BlockHash => Boolean): F[Seq[(BlockHash, BlockMessage)]] =
     for {
       _     <- metricsF.incrementCounter("find")
       state <- refF.get
-    } yield state.filterKeys(p(_)).toSeq
+    } yield state.filterKeys(p(_)).toSeq.map {
+      case (bh, bmp) => (bh, BlockMessage.from(bmp).right.get) //TODO FIX-ME
```

Comment:
same comment as with `FileLMDBIndexBlockStore.scala` above

Reply:
please open a ticket for this TODO

Reply:
Done https://rchain.atlassian.net/browse/RCHAIN-3845